File: java/src/main/java/pointmanagement/PointCluster.java
package pointmanagement;

import java.util.ArrayList;

/**
 * All points inside this cluster have the same relevant particles.
 */
public class PointCluster {

    private ArrayList<Point> insidePoints;

    PointCluster(int initialCapacity) {
        insidePoints = new ArrayList<>(initialCapacity);
    }

    public Iterable<Point> getPointsInside() {
        return insidePoints;
    }

    void add(Point point) {
        this.insidePoints.add(point);
    }

    void clear() {
        insidePoints.clear();
    }
}

File: flutter/lib/pointmanagement/point_cluster.dart
import 'point.dart';

class PointCluster {
  List<Point> _insidePoints;

  PointCluster() {
    _insidePoints = List();
  }

  List<Point> getPointsInside() {
    return _insidePoints;
  }

  void add(Point point) {
    _insidePoints.add(point);
  }

  void clear() {
    _insidePoints.clear();
  }
}